Content.Server/Administration/Managers/BanEventArgs.cs

```diff
@@ -0,0 +1,39 @@
+using Content.Shared.Database;
+using Robust.Shared.Network;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Content.Server.Administration.Managers
+{
+    public class BanEventArgs
+    {
+        public class ServerBanEventArgs : EventArgs
+        {
+            public NetUserId? Target { get; }
+            public string? TargetUsername { get; }
+            public NetUserId? BanningAdmin { get; }
+            public (IPAddress, int)? AddressRange { get; }
+            public ImmutableArray<byte>? Hwid { get; }
+            public uint? Minutes { get; }
+            public NoteSeverity Severity { get; }
+            public string Reason { get; }
+
+            public ServerBanEventArgs(NetUserId? target, string? targetUsername, NetUserId? banningAdmin, (IPAddress, int)? addressRange, ImmutableArray<byte>? hwid, uint? minutes, NoteSeverity severity, string reason)
+            {
+                Target = target;
+                TargetUsername = targetUsername;
+                BanningAdmin = banningAdmin;
+                AddressRange = addressRange;
+                Hwid = hwid;
+                Minutes = minutes;
+                Severity = severity;
+                Reason = reason;
+            }
+        }
+    }
+}
```

Comment:
This looks hella copy-pasted, you sure you can't just pass the actual ban itself around as this will be cringe to update.

Also sealed classes for new classes it gives compiler warnings if you don't.

Reply:
Indeed, it is... 
Fair enough.

I thought of iterating... but I didn't go with it, since I thought that would need a lot of function calls per player. 
I guess there is a function that check if player is banned by NetUserId or simillar.

Reply:
So... uhh... about iterating. I've found a DB function that checks if a NetUserId/IP Adress/Hardware has an active ban. But the only way (that I found) to get the NetUserId with a EntityUid is through an ActorComponent, which is removed when the player leaves the body... so... I am not sure what to do here...

Reply:
Okay, I thought about this a little bit, and I will probably store the player's NetUserId when it writes a message.
Should be good enough.